pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -3264,6 +3299,17 @@ impl Timeline {
         self.check_compaction_space(&job_desc.selected_layers)
             .await?;
 
+        self.check_memory_usage(&job_desc.selected_layers).await?;
+        if job_desc.selected_layers.len() > 100
+            && job_desc.rewrite_layers.len() as f64 >= job_desc.selected_layers.len() as f64 * 0.7
+        {
+            return Err(CompactionError::Other(anyhow!(
```

Comment:
followup: can we move this earlier to select fewer layers to rewrite, instead of failing?

Reply:
I have documented it as a current limitation and will address in the next phase of the gc-compaction work: https://github.com/neondatabase/neon/pull/8425